Save demographics info to primary applicant #162933550

akegan opened the pull request:
https://www.pivotaltracker.com/story/show/162933550

akegan committed:
[#162933550] Save demographics info to primary applicant

akegan commented on app/javascript/components/mappers/domainToApi/applicationShape.js:
```diff
@@ -21,12 +22,15 @@ export const applicationShape = {
     ])
+  mappedApplication['primaryApplicant'] = extend(mappedApplication['primaryApplicant'], mappedApplication['demographics'])
+  delete mappedApplication['demographics']
+  return mappedApplication
```
it looks like we didn't have any testing around this function at all 🤦‍♀️

akegan replied:
I've added testing around this mapping now

akegan committed:
[#162933550] Add test for primaryApplicant mapping

akegan committed:
[#162933550] Comment out E2E test that relies on short form snapshots

akegan commented:
This is ready for review. I added a quick test to verify that primaryApplicant was being mapped correctly.

In doing so, I discovered that we were putting the app member id as the application ID. Salesforce didn't seem to mind either way, so I updated the key to be factually correct. 

I've also commented out the E2E test that relies on snapshots existing. Once tests pass and the PR is removed, We can remove the commenting-out and force merge.

akegan commented on spec/javascript/components/mappers/domainToApi.test.js:
```diff
@@ -0,0 +1,71 @@
+// TODO: Expand these tests to use fixtures, test all fields in application.
```
I'd like to follow-up this PR with another one that overhauls how we do a lot of our testing

callaghanc commented:
@akegan while QA'ing this i followed the instructions as written in the pivotal story, and saw that the demographic information was saved correctly in salesforce. then i went back into the paper app i had just entered and re-saved the form. that wiped out the demographic info in salesforce. could you look into this? seems like we are not populating the demographics data on edits of paper apps.

akegan committed:
[#162933550] Show demographics on edit paper app

akegan committed:
[#162933550] Update existing snapshot tests

akegan commented:
This PR is working now, just need to add tests for mapApplication

akegan committed:
[#162933550] Update fixtures and snapshots

Add middle name, update addresses to make it clear if they are mailing addresses

akegan committed:
[#162933550] Simplify mapApplication test

Only test demographics info

akegan committed:
[#162933550] Add back confirmation modal e2e test

akegan commented on spec/javascript/fixtures/application.js:
```diff
@@ -13,8 +13,8 @@ const application = {
     "Id": "a0n0x000000AbE6AAK",
```
These fixture updates are because I started going down the path of calling mapApplication on the SOQL app then checking that it matched the domain app exactly, but abandoned that for now in favor of explicitly testing the demographics changes.

I kept the changes that I made to fixtures that make addresses clearer and add a middle name

akegan commented on spec/javascript/fixtures/short_form_submit_payload.js:
```diff
@@ -35,7 +35,7 @@ const payload = {
        phoneType: 'Cell',
        maritalStatus: 'Married',
-       applicationId: 'a0n0x000000AbE6AAK' },
+       appMemberId: 'a0n0x000000AbE6AAK' },
```
Salesforce doesn't seem to actually care what we send, but we were actually sending the appMemberId instead of the applicationId so I've renamed the field to make it clear. 

If we find later that we do in fact have to send the applicationId as part of the application member, we can add it in.

akegan committed:
[#162933550] Fix broken spec tests

Fixtures need to be updated post-refresh

akegan commented:
@callaghanc tests should be passing now. We had to make some updates to fixtures post-refresh
